api/src/main/java/org/pmiops/workbench/api/RuntimeController.java

```diff
@@ -160,16 +161,27 @@ private Runtime getOverrideFromListRuntimes(String googleProject) {
 
   @Override
   public ResponseEntity<EmptyResponse> createRuntime(String workspaceNamespace, Runtime runtime) {
-    if (runtime == null) {
-      runtime = new Runtime();
+    long configCount =
+        Stream.ofNullable(runtime)
+            .flatMap(
+                r -> Stream.of(r.getGceConfig(), r.getDataprocConfig(), r.getGceWithPdConfig()))
+            .filter(Objects::nonNull)
+            .count();
+    if (configCount != 1) {
+      throw new BadRequestException(
+          "Exactly one of GceConfig or DataprocConfig or GceWithPdConfig must be provided");
     }
 
     DbWorkspace dbWorkspace = workspaceService.lookupWorkspaceByNamespace(workspaceNamespace);
 
     GceWithPdConfig gceWithPdConfig = runtime.getGceWithPdConfig();
     if (gceWithPdConfig != null) {
       PersistentDiskRequest persistentDiskRequest = gceWithPdConfig.getPersistentDisk();
-      if (persistentDiskRequest != null && Strings.isNullOrEmpty(persistentDiskRequest.getName())) {
+      if (persistentDiskRequest == null) {
```

Comment:
new check: this condition is not valid.  We also assume that this value is not null further down, so this also prevents NPE